open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +255,454 @@ public static JSONArray appendDummyAccountID() {
+                jsonElementToDateTime.appendField(ConsentExtensionConstants.TITLE,
+                        ConsentExtensionConstants.TRANSACTION_TO_DATE_TITLE);
+                jsonElementToDateTime.appendField(StringUtils.lowerCase(ConsentExtensionConstants.DATA),
+                        toDateTimeArray);
+                consentDataJSON.add(jsonElementToDateTime);
+            }
+
+        }
+
+        /**
+         * Populate funds confirmation Details.
+         *
+         * @param initiation      data from the request
+         * @param consentDataJSON Consent information
+         */
+        private static void populateCofData(JSONObject initiation, JSONArray consentDataJSON) {
+
+            //Adding Expiration Date Time
+            if (initiation.getAsString(ConsentExtensionConstants.EXPIRATION_DATE) != null) {
+
+                if (!ConsentRetrievalUtil
+                        .validateExpiryDateTime(initiation.getAsString(ConsentExtensionConstants.EXPIRATION_DATE))) {
+                    log.error(ErrorConstants.CONSENT_EXPIRED);
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.CONSENT_EXPIRED);
+                }
+
+                String expiry = initiation.getAsString(ConsentExtensionConstants.EXPIRATION_DATE);
+                JSONArray expiryArray = new JSONArray();
+                expiryArray.add(expiry);
+
+                JSONObject jsonElementExpiry = new JSONObject();
+                jsonElementExpiry.appendField(ConsentExtensionConstants.TITLE,
+                        ConsentExtensionConstants.EXPIRATION_DATE_TITLE);
+                jsonElementExpiry.appendField(StringUtils.lowerCase(ConsentExtensionConstants.DATA), expiryArray);
+                consentDataJSON.add(jsonElementExpiry);
+            } else {
+                JSONArray expiryArray = new JSONArray();
+                expiryArray.add(ConsentExtensionConstants.OPEN_ENDED_AUTHORIZATION);
+
+                JSONObject jsonElementExpiry = new JSONObject();
+                jsonElementExpiry.appendField(ConsentExtensionConstants.TITLE,
+                        ConsentExtensionConstants.EXPIRATION_DATE_TITLE);
+                jsonElementExpiry.appendField(StringUtils.lowerCase(ConsentExtensionConstants.DATA), expiryArray);
+                consentDataJSON.add(jsonElementExpiry);
+            }
+
+            if (initiation.get(ConsentExtensionConstants.DEBTOR_ACC) != null) {
+                //Adding Debtor Account
+                populateDebtorAccount(initiation, consentDataJSON);
+            }
+        }
+
+
+        /**
+         * Populate VRP Details.
+         *
+         * @param data            Control Parameters from the request
+         * @param consentDataJSON Consent information object
+         */
+        private static void populateVRPData(JSONObject data, JSONArray consentDataJSON) {
+
+            if (data.containsKey(ConsentExtensionConstants.CONTROL_PARAMETERS)) {
```

Comment:
Shall we use negation here and throw the exception here for better readability?
```suggestion
            if (!data.containsKey(ConsentExtensionConstants.CONTROL_PARAMETERS)) {
```

Reply:
Fixed in [01f2f08]